packages/frontend/src/features/sqlRunner/runners/SqlRunnerResultsRunnerFrontend.ts

```diff
@@ -0,0 +1,234 @@
+
+const pivotQueryFn: PivotQueryFn = async ({ projectUuid, ...args }) => {
+    const scheduledJob = await schedulePivotSqlJob({
+        projectUuid,
+        ...args,
+    });
+
+    const job = await getSqlRunnerCompleteJob(scheduledJob.jobId);
+
+    if (isApiSqlRunnerJobPivotQuerySuccessResponse(job)) {
+        const url =
+            job.details && !isErrorDetails(job.details)
+                ? job.details.fileUrl
+                : undefined;
+        const results = await getResultsFromStream<RawResultRow>(url);
+
+        return {
+            results,
+            indexColumn: job.details.indexColumn,
+            valuesColumns: job.details.valuesColumns,
+        };
+    } else {
+        throw job;
+    }
+};
+
+const getSemanticLayerFieldTypeFromDimensionType = (
+    type: DimensionType,
+): SemanticLayerFieldType => {
+    switch (type) {
+        case DimensionType.STRING:
+            return SemanticLayerFieldType.STRING;
+        case DimensionType.NUMBER:
+            return SemanticLayerFieldType.NUMBER;
+        case DimensionType.BOOLEAN:
+            return SemanticLayerFieldType.BOOLEAN;
+        case DimensionType.DATE:
+        case DimensionType.TIMESTAMP:
+            return SemanticLayerFieldType.TIME;
+        default:
+            return assertUnreachable(type, `Unknown field type: ${type}`);
+    }
+};
+
+// TODO: REMOVE THIS - temporary mapping logic - also needs access to fields :(
+const convertSemanticLayerQueryToSqlRunnerPivotQuery = (
+    query: SemanticLayerQuery,
+    fields: SemanticLayerField[],
+): Pick<
+    SqlRunnerPivotQueryBody,
+    'indexColumn' | 'groupByColumns' | 'valuesColumns'
+> => {
+    const index = fields.find((field) => field.name === query.pivot?.index[0]);
+    const values = query.pivot?.values.map((value) => {
+        const customMetric = query.customMetrics?.find(
+            (metric) => metric.name === value,
+        );
+        if (!customMetric) {
+            throw new Error('Unexpected error: incorrect pivot configuration');
+        }
+        return customMetric;
+    });
+    const groupBy = query.pivot?.on.map((on) => {
+        const f = fields.find((field) => field.name === on);
+        if (!f) {
+            throw new Error('Unexpected error: incorrect pivot configuration');
+        }
+        return f;
+    });
+
+    if (index === undefined || values === undefined || values.length === 0) {
+        throw new Error('Unexpected error: incorrect pivot configuration');
+    }
+    return {
+        indexColumn: {
+            reference: index.name,
+            type: getVizIndexTypeFromSemanticLayerFieldType(index.type),
+        },
+        valuesColumns: values.map((value) => ({
+            reference: value.name,
+            aggregation: value.aggType ?? VIZ_DEFAULT_AGGREGATION,
+        })),
+        groupByColumns: groupBy?.map((f) => ({ reference: f.name })),
+    };
+};
+
+// TEMPORARY
+export const getPivotQueryFunctionForSqlRunner = ({
```

Comment:
different execution for sql runner, it mostly maps the sql runner types to a semantic layer query